### PR TITLE
CBL-521 Add revID to replicator filter callback

### PR DESCRIPTION
**Pull request by borrrden**

### CODE REVIEW

**Comment by borrrden** [comment]
I am not sure passing the c4doc here is a good idea because it does not belong to the document instance, and the document will try to free it.  That will lead to bad things.

File: Replicator/Pusher+DB.cc [hunk 1]
```diff
@@ -209,7 +209,7 @@ namespace litecore { namespace repl {
             }
 
             if (_options.pushFilter) {
-                if (!_options.pushFilter(doc->docID, doc->selectedRev.flags,
+                if (!_options.pushFilter(doc->docID, doc->revID, doc->selectedRev.flags,
```
Comment (snej) [inline Replicator/Pusher+DB.cc]: should be `doc->selectedRev.revID`

File: Replicator/Pusher.hh [hunk 2]
```diff
@@ -49,7 +49,7 @@ namespace litecore { namespace repl {
             Assert(body.buf);
             const auto bodyContent = FLValue_FromData(body, kFLTrusted);
             return _options.pushFilter
-                && _options.pushFilter(doc->docID, doc->selectedRev.flags, FLValue_AsDict(bodyContent), _options.callbackContext);
+                && _options.pushFilter(doc->docID, doc->revID, doc->selectedRev.flags, FLValue_AsDict(bodyContent), _options.callbackContext);
```
Comment (snej) [inline Replicator/Pusher.hh]: should be `doc->selectedRev.revID`

File: Replicator/Pusher+DB.cc (hunk 1 above)
Comment (borrrden) [reply Replicator/Pusher+DB.cc]: What is the difference?  I thought this was shorthand for that.

**Pull request merged**